### PR TITLE
Libuv support, data for error responses, progress for requests, refactorings

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: rockets/jsonrpc/response.h [hunk 1]
```diff
@@ -39,6 +39,7 @@ struct Response
     {
         std::string message; // short text description of the error
         int code = 0;
+        std::string data{};
```
[the remark this answers is not in the document]
Comment (tribal-tec) [reply rockets/jsonrpc/response.h]: no, is needed :)

File: tests/jsonRpcCancellable.cpp [hunk 2]
```diff
@@ -117,15 +123,14 @@ void cancelNOP()
 
 struct Fixture
 {
-    jsonrpc::AsyncReceiver jsonRpc;
+    jsonrpc::CancellableReceiver jsonRpc{[](std::string, uintptr_t) {}};
```
[the remark this answers is not in the document]
Comment (tribal-tec) [reply tests/jsonRpcCancellable.cpp]: I'll add a test

File: rockets/jsonrpc/response.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (rdumusc) [reply rockets/jsonrpc/response.h]: seriously..? Is this what allows doing Response{Error{"Invalid params", ErrorCode::invalid_params}}; without specifying the last member in the init list..?

[the remark this answers is not in the document]
Comment (tribal-tec) [reply rockets/jsonrpc/response.h]: Yes, exactly this one.

[the remark this answers is not in the document]
Comment (rdumusc) [reply rockets/jsonrpc/response.h]: OK, good to know..!

**Pull request merged**